dht/server.go

```diff
@@ -458,6 +489,11 @@ func (s *Server) announcePeer(node dHTAddr, infoHash string, port int, token str
 	if port == 0 && !impliedPort {
 		return errors.New("nothing to announce")
 	}
+	// Get node, verify that it's secure for storing announces on.
+	N := s.getNode(node, "") // May not exist, probably prone to races
```

Comment:
Yeah I don't think this is right. getNode wasn't intended to be used this way. Furthermore, the decision regarding whether to announce to a given peer should be made outside the announcePeer function. I'm not there's any value to checking whether the node we're announcing to is "secure". Security here is used to ensure that messages we're *receiving* cannot be from sources with arbitrary IDs, allowing those nodes to easily corrupt the data stored on the network.